datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,381 @@
+            ".//div[contains(@class,'tools-spec')]/div[contains(@class, 'row')]",
+        ),
+        # e.g. `Vessel information`
+        (
+            "justification_rows",
+            ".//div[contains(@class, 'tools-frame')]/div[contains(@class, 'mb-3')]",
+        ),
+        # e.g. Shipowner, Ship Safety Management Manager, Former ship names...
+        (
+            "additional_info_rows",
+            ".//div[contains(@class, 'tools-frame')]//div[@class='mb-3' or contains(@class, 'js_visibility')]",
+        ),
+    ]
+    for name, xpath_expr in xpath_definitions:
+        rows = details_container.xpath(xpath_expr)
+        for row in rows:
+            divs = row.findall("div")
+            if len(divs) == 2:
+                label_elem, value_elem = divs
+                if "yellow" in value_elem.get("class"):
+                    label, value = extract_label_value_pair(
+                        label_elem, value_elem, data
+                    )
+                    data[label] = value
+
+    web_links = details_container.xpath(
+        ".//div[contains(@class, 'tools-frame')]//a[contains(@class, 'long-text yellow')]"
+    )
+
+    name = data.pop("Vessel name (international according to IMO)")
+    type = data.pop("Vessel Type")
+    imo_num = data.pop("IMO")
+
+    vessel = context.make("Vessel")
+    vessel.id = context.make_id(name, imo_num)
+    vessel.add("name", name)
+    vessel.add("imoNumber", imo_num)
+    vessel.add("type", type)
+
+    # pop() without defaults here imply validity of the very generic selectors
+    # for these attributes.
+    vessel.add("description", " ".join(data.pop("Vessel information")))
+    vessel.add("callSign", data.pop("Call sign"))
+    vessel.add("flag", data.pop("Flag (Current)"))
+    vessel.add("mmsi", data.pop("MMSI"))
+    vessel.add("buildDate", data.pop("Build year"))
+    for raw_link in web_links:
+        link_href = raw_link.get("href", "").strip()
+        vessel.add("sourceUrl", link_href)
+    vessel.add("keywords", data.pop("Category"))
+    for pr_name in h.multi_split(data.pop("Former ship names"), [" / "]):
+        vessel.add("previousName", pr_name)
+    for past_flag in h.multi_split(data.pop("Flags (former)"), [" /"]):
+        vessel.add("pastFlags", past_flag)
+    vessel.add("topics", "poi")
+
+    sanction = h.make_sanction(context, vessel)
+    sanction.add("program", program)
+    sanction.add("sourceUrl", link)
+
+    context.emit(vessel, target=True)
+    context.emit(sanction)
+
+    linked_entity_name = data.pop(
+        "The person in connection with whom sanctions have been applied"
+    )
+    if linked_entity_name != "":
+        linked_entity = context.make("LegalEntity")
+        linked_entity.id = context.make_id(linked_entity_name)
+        for name in h.multi_split(linked_entity_name, [" | "]):
+            linked_entity.add("name", name)
+        linked_entity.add("topics", "poi")
+        context.emit(linked_entity, target=True)
```

Comment:
are we missing a relation entity for this? an UnknownLink perhaps? `role` could be `"The person in connection with whom sanctions have been applied"`

Reply:
Since these are usually entities Ian the data I said "The entity to which..."